src/load/azext_load/data_plane/load_test/custom.py

```diff
@@ -0,0 +1,287 @@
+import os
+
+import requests
+from azext_load.data_plane.utils.utils import (
+    create_or_update_body,
+    download_file,
+    get_admin_data_plane_client,
+    upload_test_plan,
+    upload_configuration_files,
+)
+from azure.cli.core.azclierror import ValidationError
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+
+def create_test(
+    cmd,
+    load_test_resource,
+    test_id,
+    display_name=None,
+    resource_group_name=None,
+    load_test_config_file=None,
+    test_description=None,
+    test_plan=None,
+    configuration_files=None,
+    engine_instances=None,
+    env=None,
+    secrets=None,
+    certificate=None,
+    key_vault_reference_identity=None,
+    subnet_id=None,
+    no_wait=False,
+):
+    client = get_admin_data_plane_client(cmd, load_test_resource, resource_group_name)
+    body = {}
+    body = create_or_update_body(
+        test_id,
+        body,
+        load_test_config_file=load_test_config_file,
+        display_name=display_name,
+        test_description=test_description,
+        engine_instances=engine_instances,
+        env=env,
+        secrets=secrets,
+        certificate=certificate,
+        key_vault_reference_identity=key_vault_reference_identity,
+        subnet_id=subnet_id,
+    )
+    list_of_tests = client.list_tests()
+    for test in list_of_tests:
+        if test_id == test.get("testId"):
+            logger.debug("Test with given test ID : %s already exists.", test_id)
+            raise ValidationError(
+                f"Test with given test ID : {test_id} already exists."
+            )
+    logger.info("Creating test with test ID: %s and body : %s", test_id, body)
+    response_obj = client.create_or_update_test(test_id=test_id, body=body)
+    logger.debug(
+        "Response object for creating test with test ID: %s is %s",
+        test_id,
+        response_obj,
+    )
+    logger.info(
+        "Created test with test ID: %s and response obj is %s", test_id, response_obj
+    )
+
+    if test_plan is not None:
+        upload_test_plan(client, test_id, test_plan, no_wait)
+
+    if configuration_files is not None:
+        upload_configuration_files(client, test_id, configuration_files)
+                    
+    return response_obj
+
+
+def update_test(
+    cmd,
+    test_id,
+    load_test_resource,
+    display_name=None,
+    resource_group_name=None,
+    load_test_config_file=None,
+    test_description=None,
+    test_plan=None,
+    configuration_files=None,
+    engine_instances=None,
+    env=None,
+    secrets=None,
+    certificate=None,
+    key_vault_reference_identity=None,
+    subnet_id=None,
+    no_wait=False,
+):
+    client = get_admin_data_plane_client(cmd, load_test_resource, resource_group_name)
+    body = client.get_test(test_id)
+    logger.debug("Retrieved test with test ID: %s and body : %s", test_id, body)
```

Comment:
do we throw an error on 404?

Reply:
yes, the SDK throws exception for everything except 2xx